src/lib/tls/asio/asio_error.h

```diff
@@ -23,6 +23,42 @@
 namespace Botan {
 namespace TLS {
 
+enum StreamError
+   {
+   StreamTruncated = 1
+   };
+
+class StreamCategory : public boost::system::error_category
```

Comment:
This category has only one error at the moment, but I think we need to make our own specific error for this case and cannot reuse some existing error_code.

Reply:
Because boost asio ssl is the openssl mplementation / requires openssl and that is what we want to replace.